cli/dcoscli/package/main.py

```diff
@@ -206,6 +216,171 @@ def _remove_repo(repo_name):
+
+    if not os.path.exists(package_json_path):
+        raise DCOSException("The file [{}] does not exist".format(package_json_path))
+
+    # load raw package json
+    with util.open_file(package_json_path) as pj:
+        package_raw = util.load_json(pj, keep_order=True)
+
+    # validate package json with local references
+    bundle_schema_path = "data/schemas/bundle-schema.json"
+    bundle_schema = util.load_jsons(
+        pkg_resources.resource_string("dcoscli", bundle_schema_path).decode("utf-8"))
+
+    errs = util.validate_json(package_raw, bundle_schema)
+
+    if errs:
+        raise DCOSException('Error validating package: '
+                            '[{}] does not conform to the specified schema'.format(package_json))
+
+    # resolve local references
+    package_resolved = _resolve_local_references(package_raw, package_directory)
+
+    # validate resolved package json
+    package_schema_path = "data/schemas/package-schema.json"
+    package_schema = util.load_jsons(
+        pkg_resources.resource_string("dcoscli", package_schema_path).decode("utf-8"))
+
+    errs = util.validate_json(package_resolved, package_schema)
+
+    if errs:
+        raise DCOSException('Error validating package: '
+                            'one of the references does not conform to the specified schema'.format(package_json))
+
+    # create zip file
+    with tempfile.NamedTemporaryFile() as temp_file:
+        with zipfile.ZipFile(
+                temp_file.file,
+                mode='w',
+                compression=zipfile.ZIP_DEFLATED,
+                allowZip64=True) as zip_file:
+            contents = json.dumps(package_resolved, indent=2)
+            zip_file.writestr("package.json", contents.encode())
+
+        # name the zip file appropriately
+        zip_file_name = os.path.join(
+            output_directory,
+            '{}-{}-{}.dcos'.format(
+                package_resolved['name'],
+                package_resolved['version'],
+                sha256_file(temp_file.name)))
+
+        if os.path.exists(zip_file_name):
+            raise DCOSException(
+                'Output file [{}] already exists'.format(
+                    zip_file_name))
+
+        util.sh_copy(temp_file.name, zip_file_name)
+
+    emitter.publish(
+            'Created DCOS Universe package [{}].'.format(zip_file_name))
+
+    return 0
+
+
+def _resolve_local_references(package_json, package_directory):
+    """ Resolves all local refereces in package json
+    :param package_json: The package json that may contain local references
+    :type package_json: dict
+    :param package_directory: The directory of the project.
+    :type package_directory: str
+    :returns: The package json with all references resolved
+    :rtype: dict
+    """
+    bundle_schema_path = "data/schemas/bundle-schema.json"
+    bundle_schema = util.load_jsons(
+        pkg_resources.resource_string("dcoscli", bundle_schema_path).decode("utf-8"))
```

Comment:
For example you already constructed this value from the function that calls this function.